src/groups/mqb/mqbstat/mqbstat_clusterstats.cpp

```diff
@@ -442,61 +434,71 @@ ClusterStats::setPartitionCfgBytes(bsls::Types::Int64 dataBytes,
         ClusterStatsIndex::e_PARTITION_CFG_JOURNAL_BYTES,
         journalBytes);
     d_statContext_mp->setValue(ClusterStatsIndex::e_CSL_CFG_BYTES, cslBytes);
-    bsl::vector<bsl::shared_ptr<bmqst::StatContext> >::const_iterator it =
-        d_partitionsStatContexts.cbegin();
-    for (; it != d_partitionsStatContexts.cend(); ++it) {
-        (*it)->setValue(ClusterStatsIndex::e_PARTITION_CFG_DATA_BYTES,
-                        dataBytes);
-        (*it)->setValue(ClusterStatsIndex::e_PARTITION_CFG_JOURNAL_BYTES,
-                        journalBytes);
+    bsl::vector<bsl::shared_ptr<PartitionStats> >::const_iterator it =
+        d_partitionsStats.cbegin();
+    for (; it != d_partitionsStats.cend(); ++it) {
+        (*it)->statContext()->setValue(
+            ClusterStatsIndex::e_PARTITION_CFG_DATA_BYTES,
+            dataBytes);
+        (*it)->statContext()->setValue(
+            ClusterStatsIndex::e_PARTITION_CFG_JOURNAL_BYTES,
+            journalBytes);
     }
     return *this;
 }
 
-ClusterStats& ClusterStats::setNodeRoleForPartition(int partitionId,
-                                                    PrimaryStatus::Enum value)
+// --------------------
+// class PartitionStats
+// --------------------
+
+PartitionStats::PartitionStats(
+    const bsl::shared_ptr<bmqst::StatContext>& statContext)
+: d_statContext_sp(statContext)
+{
+    // NOTHING
+}
+
+PartitionStats& PartitionStats::setRoloverTime(bsls::Types::Int64 value)
+{
+    d_statContext_sp->reportValue(ClusterStatsIndex::e_PARTITION_ROLLOVER_TIME,
+                                  value);
+    return *this;
+}
+
+PartitionStats& PartitionStats::setReplicationTime(bsls::Types::Int64 value)
```

Comment:
For better performance, you might move these setters to the header and make them inline. At least the ones that are called often (`setReplicationTime`?)